src/cbc_syslog/util/output.py

```diff
@@ -0,0 +1,118 @@
+
+# *******************************************************
+# Copyright (c) VMware, Inc. 2020-2023. All Rights Reserved.
+# SPDX-License-Identifier: MIT
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Output class"""
+
+import logging
+import requests
+import socket
+import ssl
+import traceback
+
+log = logging.getLogger(__name__)
+
+
+class Output:
+    """Output mechanisms to send data to local or external destination"""
+
+    def __init__(self, output_params):
+        """
+        Initialize the Output object.
+
+        Args:
+            output_params (dict): The output configuration
+
+        Example:
+            {
+                "type": "",
+                "host": "",
+                "port": "",
+                "tls_verify": "",
+                "http_headers": "",
+                "ca_cert": "",
+                "cert": "",
+                "key": "",
+                "key_password": ""
+            }
+        """
+        self.output_params = output_params
+
+    def send(self, data):
+        """
+        Send data to configured destination.
+
+        Args:
+            data (str): The content to be sent
+        """
+        success = True
+        client_socket = None
+
+        encoded_data = data.encode("utf-8")
+
+        type = self.output_params.get("type").lower()
+        server_url = self.output_params.get("host")
+        port = self.output_params.get("port")
+
+        if "tcp" in type:
+            unsecured_client_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            try:
+                if "tls" in type:
+                    context = ssl.create_default_context()
+                    context.load_verify_locations(self.output_params["ca_cert"])
+
+                    if "cert" in self.output_params:
+                        context.load_cert_chain(self.output_params["cert"],
+                                                keyfile=self.output_params["key"],
+                                                password=self.output_params["key_password"])
+
+                    if not self.output_params.get("tls_verify", True):
```

Comment:
Is the type of the value **string** or **bool**? Based on the example - `"tls_verify": "",`

Reply:
bool and I'll update the example for context on the types